src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResourceVersion2.java

```diff
@@ -385,6 +385,11 @@ private DataAccessRequest populateDarFromJsonString(User user, String json) {
           && existingDar.getUserId().equals(user.getUserId())
           && existingDar.getDraft()) {
         newDar.setReferenceId(data.getReferenceId());
+
+        // if draft was part of a collection, we should use the same collection.
+        if (Objects.nonNull(existingDar.getCollectionId())) {
```

Comment:
nitpick: what would you think of adding a method on the existing dar for "isDraft()" that returns `Objects.nonNull(existingDar.getCollectionId()`?

I would rec that because I dont know if the null ID will *always* indicate draft status, and it might be a nice safe way for us to update this type of check across the app quickly in the future.

Reply:
ohh to clarify this isn't checking if it is a draft, but that if the DAR which already exists in the db has a collection we should pull that id in. I will clarify the comment above; this just makes sure when the draft is resubmitted it doesn't make a new collection, but keeps the old one (since it's already there).